Sources/SourceKit/LanguageService.swift

```diff
@@ -0,0 +1,45 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import LanguageServerProtocol
+
+/// A `LanguageServer` that exists within the context of the current process.
+public protocol LanguageService: AnyObject {
```

Comment:
I've been thinking about exposing an API along these lines to make it more convenient to write code agains the LSP server directly, but I think for this particular change we're better off with a special-purpose API used for talking to the swift and clang language implementations. For example, I would argue `documentUpdatedBuildSettings` and probably `symbolInfo` are more relevant to the implementation.  So I suggest naming this something like `ToolchainLanguageServer`, and moving it into the `SourceKit` module.  We might in the future add back something like this, but I think it serves a somewhat different purpose than the API we want for talking to the toolchain server.

Reply:
Renamed, but it's already in the SourceKit module